src/misc/setup.cpp

```diff
@@ -832,11 +845,13 @@ Section_prop *Config::AddEarlySectionProp(const char *name,
 	return s;
 }
 
-Section_prop *Config::AddSection_prop(char const *const name,
+Section_prop *Config::AddSection_prop(const char *section_name,
                                       SectionFunction func,
                                       bool changeable_at_runtime)
 {
-	Section_prop *s = new Section_prop(name);
+	assertm(std::regex_match(section_name, std::regex{"[a-zA-Z]+"}),
```

Comment:
> Anyone getting this error in newest master? dosbox: setup.cpp:853: Section_prop* Config::AddSection_prop(const char*, SectionFunction, bool): Assertion ((void)"Only letters are allowed in section name", std::regex_match(section_name, std::regex{"[a-zA-Z]+"}))' failed.

@dreamer , this is a debug build.  Also reports that this triggers before `dosbox --help`.  
I can't remember if this asserts against user-side content (such as .conf file content). If so, we should probably switch to non-assert and provide our usual helpful error and offending conf key or value.

Reply:
Addressed in 19e5563fa4bca00d4d36d2c5056d2cb74ef22888 - we didn't catch this via CI as MT-32 is still off by default and somehow missed it in the review as well. Good that it's fixed now - thanks for letting me know! :)